packages/grafana-ui/src/components/Combobox/MultiCombobox.tsx

```diff
@@ -48,8 +43,11 @@ export const MultiCombobox = <T extends string | number>(props: MultiComboboxPro
   }, [value, options, isAsync]);
 
   const styles = useStyles2(getComboboxStyles);
+  const [inputValue, setInputValue] = useState('');
+
+  const [baseItems, baseSetItems] = useState(isAsync ? [] : options);
 
-  const [items, baseSetItems] = useState(isAsync ? [] : options);
+  const items = useMemo(() => baseItems.filter(itemFilter(inputValue)), [baseItems, inputValue]);
```

Comment:
Should we be filtering if we use the promise API? The `options` param is getting input as an arg and so I would assume it should handle filtering itself. Should we still filter afterwards in that case?

For example the `options` function can implement wild cards because it could use a specific backend, then we would filter just based on a text match here possibly filtering more than it should.

Reply:
This will be handled when we implement the async behavior.